pkg/reconciler/service/service.go

```diff
@@ -309,31 +309,39 @@ func (c *Reconciler) createConfiguration(service *v1alpha1.Service) (*v1alpha1.C
 	return c.ServingClientSet.ServingV1alpha1().Configurations(service.Namespace).Create(cfg)
 }
 
-func configSemanticEquals(desiredConfig, config *v1alpha1.Configuration) bool {
+func configSemanticEquals(ctx context.Context, desiredConfig, config *v1alpha1.Configuration) (bool, error) {
+	logger := logging.FromContext(ctx)
+	specDiff, err := kmp.SafeDiff(desiredConfig.Spec, config.Spec)
```

Comment:
Why not `SafeEqual`? We don't need the diff output here and it takes more time to generate the diff.

Reply:
the diff has been there from before. In case of a mismatch it is helpful to know what diff is causing error in reconciliation (hence logging it).

+1 on the e2e test.

Reply:
> what diff is causing error in reconciliation (hence logging it).

Has this actually been useful in practice? The log statement happens on every reconcile loop and the diff is only output when the reconciler detects it needs to take action (not on error).

Reply:
yeah sorry I meant to say it helps with figuring out what diff is causing reconciliation.

I dont have too many data points on whether or not this has been helpful in the past. But certainly as @taragu was doing the development for initial scale zero, it helped her figure out some failures in the e2es and how the reconciler was trying to reconcile old and new configuration and route.

Reply:
Sounds good. Let's keep it for now then.